node-server/src/services/message-handler.js

```diff
@@ -0,0 +1,107 @@
+/**
+ * Handles synchronization messages recieved from clients and other servers
+ * @typedef {import(../../../common/types/datatypes).SyncMessage} SyncMessage
+ */
+
+const logger    = require('../../../common/utils/logger')
+const db = require('./database')
+const {
+  getNodeId
+} = require('../state/node')
+let synchronizer = null
+
+const installSynchronizer = (synchronizerObj) => {
+  synchronizer = synchronizerObj
+}
+
+/**
+ * Makes database querys and returns promises
+ * @param {*} object
+ * @returns {Promise<*>}
+ * @private
+ */
+/* eslint-disable no-case-declarations */
+const handle = async (address, object) => {
+  if (!synchronizer) {
+    logger.error('SYNCHRONIZER not installed')
+  }
+
+  if (object.charAt(0) === '{') {
+    /** @type  */
+    const message = JSON.parse(object)
+    switch (message.name) {
+      case 'syncRequest':
+        const diff = await synchronizer.getMessageDiff(message.payload)
+        logger.info(`Sync request received from ${address}: ${message}`)
+        return JSON.stringify({ name: 'syncReply', payload: diff })
+
+      case 'synchReply':
```

Comment:
Typo:

```suggestion
      case 'syncReply':
```

This currently breaks the syncing.